senza/cli.py

```diff
@@ -1449,28 +1459,23 @@ def wait(stack_ref, region, deletion, timeout, interval,):
                             'status {}'.format(stack.name,
                                                stack.version,
                                                stack.StackStatus))
-            else:
-                stacks_nok.add((stack.name, stack.version, stack.StackStatus))
 
-        if stacks_nok:
+        if stacks_in_progress:
             waiting_for = ', '.join(['{}-{} ({})'.format(*x)
-                                     for x in sorted(stacks_nok)])
+                                     for x in sorted(stacks_in_progress)])
             info('Waiting up to {:.0f} more secs '
                  'for stack{} {}..'.format(cutoff - time.time(),
-                                           's' if len(stacks_nok) > 1 else '',
+                                           's' if len(stacks_in_progress) > 1 else '',
                                            waiting_for))
-        elif stacks_ok:
+            time.sleep(interval)
+            continue
+        if stacks_ok:
             successful_stacks = ', '.join(['{}-{}'.format(*x)
                                            for x in sorted(stacks_ok)])
             actions = '/'.join(successful_actions)
             ok('OK: Stack(s) {} {} successfully.'.format(successful_stacks,
                                                          actions))
             return
-        else:
-            raise click.UsageError('No matching stack for '
-                                   '"{}" found'.format(' '.join(stack_ref)))
-        time.sleep(interval)
-    raise click.Abort()
```

Comment:
removing the `click.Abort` would result in exit code 0 if the timeout (`cutoff`) is reached